storage/innobase/log/log0recv.cc

```diff
@@ -1793,7 +1793,7 @@ recv_parse_or_apply_log_rec_body(
 		of offline backup and continue.
 		*/
 		if (!recv_recovery_on) {
-			if (!opt_lock_ddl_per_table) {
+			if (mdl_taken) {
 				if (backup_redo_log_flushed_lsn
```

Comment:
Please add the reason you mentioned on slack as code comment. Why a bool is better to have than removing it

Reply:
I would prefer that we keep it for two reasons:

1. If you have a huge list of tables, querying INNODB_SYS_TABLES can take sometime, and there is no problem in accepting MLOG_INDEX_LOAD while we are not done with the MDL step.
2. In case PXB is waiting on MDL like an ALTER table is ongoing, it's also safe to keep accepting MLOG_INDEX_LOAD on tables that we have not yet acquired MDL